CIPs/CIP-85/CIP-85.md

```diff
@@ -50,116 +50,121 @@ A Collection "instance" would therefore be made of 1 `AppendCollection` document
-    last: { type: 'string', maxLength: 150 },
   },
-  required: ['sliceSize', 'slicesCount', 'first', 'last']
+  required: ['sliceMaxItems', 'slicesCount']
 }
 ```
 
 #### CollectionSlice schema
 
 The `CollectionSlice` schema must be an `object` with a `$ceramic` field from [CIP-88](../CIP-88/CIP-88.md) having the type `collectionSlice` and pointing to the slice's `schema`, along with the following properties:
 
-- `contents`: array with a `maxItems` value matching the `AppendCollection` `sliceSize` property and defining the items schemas, that must include `{ type: 'null' }` in order to support removals
-- `next`: DocID of the next slice (if the slice is not the last one)
-- `prev`: DocID of the previous slice (if the slice is not the first one)
+- `collection`: the DocID of the collection the slice is part of
+- `sliceIndex`: index of the slice in the collection, between `0` and the collection's `slicesCount` minus `1`
+- `contents`: array with a `maxItems` value matching the `AppendCollection` `sliceMaxItems` property and defining the items schemas, that must include `{ type: 'null' }` in order to support removals
 
 ```js
 {
   $schema: 'http://json-schema.org/draft-07/schema#',
-  $ceramic: { type: 'collectionSlice', schema: '<slice schema ID>' },
+  $ceramic: { type: 'collectionSlice' },
   title: 'MyCollectionSLice',
   type: 'object',
   properties: {
+    collection: { type: 'string', maxLength: 150 },
+    sliceIndex: { type: 'integer', minimum: 0 },
     contents: {
       type: 'array',
       maxItems: 256,
-      minItems: 1,
+      minItems: 0,
       items: {
         oneOf: [{ type: 'object', ... }, { type: 'null' }],
       },
     },
-    next: { type: 'string', maxLength: 150 },
-    prev: { type: 'string', maxLength: 150 },
   },
-  required: ['contents'],
+  required: ['collection', 'sliceIndex', 'contents'],
 }
 ```
 
 ### Algorithms
 
+#### Deterministic slice access
+
+Accessing slices in the collection relies on Ceramic's ability to load documents deterministically based on their genesis contents:
+
+- `collection`: the collection's `DocID` string
+- `sliceIndex`: the slice's index, an integer between `0` (first slice) and `slicesCount - 1` (last slice)
+- `contents`: empty array
+
 #### First insertion
 
 > This flow assumes a prerequisite check that the `AppendCollection` document has not been created yet.
 
-1. Create the `CollectionSlice` document with the `contents` array containing the item to insert.
-1. Create the `AppendCollection` document with a `slicesCount` of `1`, and the `first` and `last` values containing the DocID of the created `CollectionSlice`.
+1. Create the `AppendCollection` document.
+1. Create a deterministic `CollectionSlice` document with `sliceIndex` of `0` and an empty `contents` array.
+1. Update the created `CollectionSlice` document with the `contents` array containing the item to insert.
+1. Create the `AppendCollection` document with a `slicesCount` of `1`.
```

Comment:
```suggestion
1. Update the `AppendCollection` document with a `slicesCount` of `1`.
```

Reply:
This step can then be removed entirely if the first step is to create the `AppendCollection` doc with a slicesCount of 1